server.js

```diff
@@ -0,0 +1,237 @@
+const http = require('http');
+const fs = require('fs');
+const querystring = require('querystring');
+
+const PORT = 8080;
+
+const server = http.createServer((req, res) => {
+  let method = req.method;
+  let URL = req.url;
+  let headers = req.headers;
+      if (method === 'GET'){
+        if (URL === '/'){
+          URL = '/index.html';
+        }
+        splitURL = URL.split('.');
+        let type = splitURL[1];
+        let body = '';
+        req.on('data', (chunk) => {
+          body += chunk;
+        });
+        fs.readdir('./public', (err, dir) => {
```

Comment:
This should be in the req.on('end') to make sure that you've fully read the body before trying to respond to it.